modules/Bio/EnsEMBL/Hive/DBSQL/StatementHandle.pm

```diff
@@ -179,4 +198,29 @@ sub DESTROY {   # note AUTOLOAD/DESTROY interdependence!
     }
 }
 
+
+## Just like AUTOLOAD for function calls, we need to redirect
+## the HASH methods to the DBI::st instance
+
+package DBIstHashProxy;
+
+use base ('Tie::Hash');
+
+sub TIEHASH {
+    my $class = shift;
+    my $self = $_[0];
+    bless $self, $class;
+    return $self;
+}
+
+sub STORE    { $_[0]->{'_dbi_sth'}->{$_[1]} = $_[2] }
+sub FETCH    { $_[0]->{'_dbi_sth'}->{$_[1]} }
+sub FIRSTKEY { my $a = scalar keys %{$_[0]->{'_dbi_sth'}}; each %{$_[0]->{'_dbi_sth'}} }
```

Comment:
What does $a do here?

Reply:
Actually, I don't really know. I've copied it from the default implementation of Tie::StdHash, see /nfs/software/ensembl/RHEL7-JUL2017-core2/plenv/versions/5.24.1/lib/perl5/5.24.1/Tie/Hash.pm lines 249-256 (I can't find it online !)
My guess is is that's it's there to reset the internal cursor.